### PR TITLE
Massmint onboarding

### LINKED ISSUE #5176: MassMint: Onboarding
* #4964 

## Figma
---
- [Detailed interactions and logic is available Figma](https://www.figma.com/file/HCmIzbN0a2XhsOstHkLiwi/Mass-Mint---Handoff?node-id=0%3A1&t=ynDINS3WQGGG9Dj0-1). 
- all the needed screens
- [interactive prototype ](https://www.figma.com/proto/HCmIzbN0a2XhsOstHkLiwi/Mass-Mint---Handoff?page-id=0%3A1&node-id=1%3A332&viewport=1014%2C587%2C0.04&scaling=scale-down-width&starting-point-node-id=1%3A332&show-proto-sidebar=1)

Please take the prototype with a grain of salt, not everything is perfect and 1:1 to final design how it should look, so in case you dont know something, dont hesitate to ask @VladoTheBoi 

## Onboarding
- onboarding is displayed only once when  entering Mass Mint for the first time, 
   when entering for the 2nd time, no onboarding displayed  , user is taken straight to mass mint
- user can scroll between slides by pressing Next button or Arrows 
- user has option to go back to onboarding from mass mint feature
- button “done” is displayed on the last slide
- “skip” button is displayed on every slide except the last one 
image

@VladoTheBoi  how to navigate to this page?

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #5176 
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [x] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image
image
image

### CODE REVIEW

**Comment by daiagi** [comment]
as for the deepSource issue:

> > what should i do to fix it?
> 
> ignore them :see_no_evil: (https://github.com/kodadot/nft-gallery/pull/5205#issuecomment-1460304222)

**Review by roiLeo — COMMENTED** [review]
- `store/massmint.ts` should be move in `preferences` store (no pinia yet I'm on it)
- scss file indentation

**Comment by daiagi** [comment]
> - `store/massmint.ts` should be move in `preferences` store (no pinia yet I'm on it)
> - scss file indentation 

Yes I was thinking the same.
Waiting on you to transform it to pinia tbh

File: components/massmint/OnBoarding.vue [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        :label="btn.label"
+        class="is-flex-grow-1 limit-width"
+        :variant="btn.variant"
+        @click.native="btn.onClick" />
+    </div>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { ref } from 'vue'
```
Comment (Jarsen136) [inline components/massmint/OnBoarding.vue:69]:
```suggestion
```

The `ref` would be auto imported

File: components/massmint/OnBoarding.vue [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+      path: `/${urlPrefix.value}/massmint`,
+    })
+    .catch($consola.warn)
+}
+
+const btn = computed(() =>
+  currentSlide.value === 2
+    ? {
+        label: $i18n.t('massmint.onboarding.done'),
+        variant: 'k-accent' as NeoButtonVariant,
```
Comment (Jarsen136) [inline components/massmint/OnBoarding.vue:118]: `as NeoButtonVariant`
Is it necessary to have this

File: components/massmint/OnBoarding.vue [hunk 3]
```diff
@@ -0,0 +1,250 @@
...
+        onClick: nextSlide,
+      }
+)
+
+const onTouchStart = (event: TouchEvent) => {
+  startX.value = event.touches[0].clientX
+  startY.value = event.touches[0].clientY
+}
+
+const onTouchMove = (event: TouchEvent) => {
```
Comment (Jarsen136) [inline components/massmint/OnBoarding.vue:133]: Let's try to use `usedraggable` from `vueuse` https://vueuse.org/core/useDraggable/#usedraggable
Then I guess we could remove the `onTouchStart` and `onTouchMove` event listeners.

File: components/massmint/OnBoarding.vue (hunk 2 above)
Comment (daiagi) [reply components/massmint/OnBoarding.vue:117]: image


```Type 'string' is not assignable to type 'NeoButtonVariant | undefined'```

File: components/massmint/OnBoarding.vue (hunk 3 above)
Comment (daiagi) [reply components/massmint/OnBoarding.vue]: useSwipe turned out be more suitable.

File: components/navbar/CreateDropdown.vue [hunk 4]
```diff
@@ -16,6 +16,17 @@
...
+      <b-dropdown-item has-link>
+        <b-tooltip
+          position="is-left"
+          :label="$t('createDropdown.massmint')"
+          class="navbar-item-tooltip">
+          <nuxt-link data-cy="massmint" :to="`/${urlPrefix}/massmint`">
+            {{ $t('multipleNFTS') }}
+          </nuxt-link>
+        </b-tooltip>
+      </b-dropdown-item>
```
Comment (Jarsen136) [inline components/navbar/CreateDropdown.vue:28]: The option should be visible only on the `redesign` mode. Let's make the url parms to `?redesign=true` to see this nav item.

Comment (daiagi) [reply components/navbar/CreateDropdown.vue:28]: Sure thing

Comment (daiagi) [reply components/navbar/CreateDropdown.vue]: done

**Review by Jarsen136 — APPROVED** [review]
code lgtm

**Comment by daiagi** [comment]
> - `store/massmint.ts` should be move in `preferences` store (no pinia yet I'm on it)

@roiLeo should I use preferences store as is right now or wait for a proper fix?

**Comment by roiLeo** [comment]
> > * `store/massmint.ts` should be move in `preferences` store (no pinia yet I'm on it)
> 
> @roiLeo should I use preferences store as is right now or wait for a proper fix?

If you need to store basic user preference then yes! 
But if you will need to write specific use case for massMint (not only one getter/setter) then I would keep this new file. 

>  or wait for a proper fix?

What's wrong again? only https://github.com/kodadot/nft-gallery/pull/5246#issuecomment-1465980097 seems weird to me but I didn't really take the time to look at

**Comment by daiagi** [comment]
> What's wrong again? only [#5246 (comment)](https://github.com/kodadot/nft-gallery/pull/5246#issuecomment-1465980097) seems weird to me but I didn't really take the time to look at

@roiLeo  the actions were mistyped. see #5253

**Review by roiLeo — APPROVED** [review]
Nice! cards are swipeable on mobile!

This page is empty for now.
Screenshot 2023-03-14 at 08-40-14 Low minting fees and carbonless NFTs KodaDot - NFT Market Explorer

maybe it would be cool to allow user to click on dot

Otherwise wfm ✅

**Pull request merged**